src/api/unified/symbol_manager.hpp

```diff
@@ -133,16 +135,21 @@ bool checkArrays(af_backend activeBackend, T a, Args... arg) {
                             AF_ERR_ARR_BKND_MISMATCH);                        \
     } while (0)
 
-#if defined(OS_WIN)
-#define CALL(...) \
-    unified::AFSymbolManager::getInstance().call(__FUNCTION__, __VA_ARGS__)
-#define CALL_NO_PARAMS() \
-    unified::AFSymbolManager::getInstance().call(__FUNCTION__)
-#else
-#define CALL(...) \
-    unified::AFSymbolManager::getInstance().call(__func__, __VA_ARGS__)
-#define CALL_NO_PARAMS() unified::AFSymbolManager::getInstance().call(__func__)
-#endif
+#define CALL(FUNCTION, ...)                                                   \
+    using af_func = std::add_pointer<decltype(FUNCTION)>::type;               \
+    static af_backend index_ =                                                \
+        unified::AFSymbolManager::getInstance().getActiveBackend();           \
+    static af_func func = (af_func)common::getFunctionPointer(                \
+        unified::AFSymbolManager::getInstance().getHandle(), __func__);       \
+    if (index_ !=                                                             \
+        unified::AFSymbolManager::getInstance().getActiveBackend()) {         \
+        index_ = unified::AFSymbolManager::getInstance().getActiveBackend();  \
```

Comment:
I understand initialization of `index_` is not an issue, but what happens when more than one thread call the same line of code ? This feels more like a `thread_local` instead of just static.